browser/ui/tabs/brave_tab_color_mixer.cc

```diff
@@ -8,19 +8,53 @@
 #include "base/containers/fixed_flat_map.h"
 #include "brave/browser/ui/color/brave_color_id.h"
 #include "brave/ui/color/nala/nala_color_id.h"
+#include "chrome/browser/themes/theme_properties.h"
 #include "chrome/browser/ui/color/chrome_color_id.h"
 #include "ui/color/color_mixer.h"
 #include "ui/color/color_provider.h"
 #include "ui/color/color_provider_key.h"
 #include "ui/color/color_recipe.h"
 #include "ui/color/color_transform.h"
+#include "ui/gfx/color_utils.h"
 
 namespace tabs {
 
+SkColor IsToolbarColorDark(const ui::ColorProviderKey& key,
+                           const ui::ColorMixer& mixer) {
+  CHECK(key.custom_theme);
+  auto toolbar_color = mixer.GetResultColor(kColorToolbar);
+  SkColor custom_toolbar_color;
+  if (key.custom_theme->GetColor(ThemeProperties::COLOR_TOOLBAR,
+                                 &custom_toolbar_color)) {
+    toolbar_color = custom_toolbar_color;
+  }
+
+  return color_utils::IsDark(toolbar_color);
+}
+
 void AddBraveTabThemeColorMixer(ui::ColorProvider* provider,
                                 const ui::ColorProviderKey& key) {
   auto& mixer = provider->AddMixer();
 
+  if (key.custom_theme) {
+    const bool is_toolbar_dark = IsToolbarColorDark(key, mixer);
+    mixer[kColorBraveSplitViewTileBackgroundHorizontal] = {
+        is_toolbar_dark ? nala::kColorPrimitiveNeutral10
+                        : nala::kColorPrimitiveNeutral80};
+    mixer[kColorBraveSplitViewTileBackgroundVertical] = {
+        kColorBraveSplitViewTileBackgroundHorizontal};
+    mixer[kColorBraveSplitViewTileDivider] = {
+        is_toolbar_dark ? nala::kColorPrimitiveNeutral20
+                        : nala::kColorPrimitiveNeutral90};
```

Comment:
As these colors are our own color, custom theme doesn't have color for them.
@aguscruiz what I test here is picking similar color of toolbar between our light/dark theme colors for split view tile bg color.
You can see the some screenshots with downloaded custom theme from webstore.
Or, maybe we need to generate when custom theme is used?

Reply:
<img width="835" alt="image" src="https://github.com/user-attachments/assets/6349d03e-81b2-4811-be56-3b729bf122e6" />

In these two screenshots the theme is dark but the color being used for active tabs and split tabs are the light ones. So, in those cases, the color theme the user chose independent of wether they have light/dark mode in the browser. Just tested it myself to confirm that. 

Let me see how I can solve that split view bg being bad. We'll need a custom solution here, like you suggested